### PR TITLE
Feature/backtracking path

### PR DESCRIPTION
**Pull request by TBouder**
## What it does ✨
First part for a larger update. This PR add a new folder `scripts` with a script used to find, from one CRV vault, the path to another CRV vault. This script is not meant to be fast, it should mainly work and find the correct path, if a path exists.
The [Curve documentation](https://curve.readthedocs.io/ref-addresses.html) is used for deployment addresses.
The vaults used for this script are the following : 
```js
const	VAULTS = [
	['crvALUSD', '0xA74d4B67b3368E83797a35382AFB776bAAE4F5C8'],
	['crvSBTC', '0x8414Db07a7F743dEbaFb402070AB01a4E0d2E45e'],
	['crvGUSD', '0x2a38B9B0201Ca39B17B460eD2f11e4929559071E'],
	['crvSUSD', '0x5a770DbD3Ee6bAF2802D29a901Ef11501C44797A'],
	['crvUST', '0x1C6a9783F812b3Af3aBbf7de64c3cD7CC7D1af44'],
	['crvLINK', '0xf2db9a7c0ACd427A680D640F02d90f6186E71725'],
	['3crv', '0x84E13785B5a27879921D6F685f041421C7F482dA'],
	['crvAETHc', '0x132d8D2C76Db3812403431fAcB00F3453Fc42125'],
	['crvSTETH', '0xdCD90C7f6324cfa40d7169ef80b12031770B4325'],
	['crvSETH', '0x986b4AFF588a109c09B50A03f42E4110E29D353F'],
	['crvBUSD', '0x6Ede7F19df5df6EF23bD5B9CeDb651580Bdf56Ca'],
	['crvRETH', '0xBfedbcbe27171C418CDabC2477042554b1904857'],
	['crvUSDN', '0x3B96d491f067912D18563d56858Ba7d6EC67a6fa'],
	['crvRSV', '0xC116dF49c02c5fD147DE25Baa105322ebF26Bd97'],
	['yCRV', '0x4B5BfD52124784745c1071dcB244C6688d2533d3'],
	['crvPBTC', '0x3c5DF3077BcF800640B5DAE8c91106575a4826E6'],
	['crvHUSD', '0x054AF22E1519b020516D72D749221c24756385C9'],
	['crvSAAVe', '0xb4D1Be44BfF40ad6e506edf43156577a3f8672eC'],
	['crvRENBTC', '0x7047F90229a057C13BF847C0744D646CFb6c9E1A'],
	['crvOBTC', '0xe9Dc63083c464d6EDcCFf23444fF3CFc6886f6FB'],
	['crvAAVE', '0x39CAF13a104FF567f71fd2A4c68C026FDB6E740B'],
	['crvTUSD', '0xf8768814b88281DE4F532a3beEfA5b85B69b9324'],
	['crvDUSD', '0x30FCf7c6cDfC46eC237783D94Fc78553E79d4E9C'],
	['crvHBTC', '0x625b7DF2fa8aBe21B0A976736CDa4775523aeD1E'],
	['crvLUSD', '0x5fA5B62c8AF877CB37031e0a3B2f34A78e3C56A6'],
	['crvUSDK', '0x3D27705c64213A5DcD9D26880c1BcFa72d5b6B0E'],
	['crvFRAX', '0xB4AdA607B9d6b2c9Ee07A275e9616B84AC560139'],
	['crvTriCrypto', '0x3D980E50508CFd41a13837A60149927a11c03731'],
	['crvCOMP', '0xD6Ea40597Be05c201845c0bFd2e96A60bACde267'],
	['crvPAX', '0x80bbeE2fa460dA291e796B9045e93d19eF948C6A'],
	['crvBBTC', '0x8fA3A9ecd9EFb07A8CE90A6eb014CF3c0E3B32Ef'],
	['crvTBTC', '0x23D3D0f1c697247d5e0a9efB37d8b0ED0C464f7f'],
	['crvMUSD', '0x8cc94ccd0f3841a468184aCA3Cc478D2148E1757'],
	['crvYBUSD', '0x8ee57c05741aA9DB947A744E713C15d4d19D8822'],
	['crvID', '0x27b7b1ad7288079A66d12350c828D3C00A6F07d7'],
	['crv3Crypto', '0xE537B5cc158EB71037D4125BDD7538421981E6AA'],
	['crvEURs', '0x25212Df29073FfFA7A67399AcEfC2dd75a831A1A'],
	['crvUSDP', '0xC4dAf3b5e2A9e93861c3FBDd25f1e943B8D87417'],
];
```

## How to test ✅
By running `node scripts/findPath.js > temp.txt` a list of path should be found. With theses paths, you should be able to successfully run the test script from the Bowswap smartcontract. 
There is currently 450+ paths, and the script has a limitation of 30/min, so the path should be splitted (or via VPN).  
There is currently some issues with the `crvUSDK` vault (no liquidity to perform the test).  
There is currently an issue with some specific path because of old Curve implementation & incompatibility with the current version of the SC.

### CODE REVIEW

File: scripts/findPath.js [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+		if (result !== false) {
+			returnValue = result;
+			break;
+		}
+	}
+	return returnValue;
+}
+
+async function	findAllPath() {
+	const	VAULT_ADDRESSES = [
```
Comment (fameal) [inline scripts/findPath.js:199]: Do we really need these addresses hardcoded? Is there any benefit putting them in a JSON file? Is this list related to the list in #7 ? or LISTING.json?

File: scripts/findPath.js [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+		for (let secondVaultIndex = 0; secondVaultIndex < VAULT_ADDRESSES.length; secondVaultIndex++) {
+			const secondElement = VAULT_ADDRESSES[secondVaultIndex];
+			if (element === secondElement) {
+				continue;
+			}
+			const	result = await findPath({from: element, to: secondElement});
+			if (result) {
+				console.dir([element, secondElement, [...result]]);
+			} else {
+				// console.dir([element, secondElement, [[]]]);
```
Comment (fameal) [inline scripts/findPath.js:251]: Remove this just to keep it clean, unless you need to keep it for debugging

File: scripts/findPath.js [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+	for (let vaultIndex = 0; vaultIndex < VAULT_ADDRESSES.length; vaultIndex++) {
+		const element = VAULT_ADDRESSES[vaultIndex];
+		for (let secondVaultIndex = 0; secondVaultIndex < VAULT_ADDRESSES.length; secondVaultIndex++) {
+			const secondElement = VAULT_ADDRESSES[secondVaultIndex];
+			if (element === secondElement) {
+				continue;
+			}
+			const	result = await findPath({from: element, to: secondElement});
+			if (result) {
```
Comment (fameal) [inline scripts/findPath.js:248]: Remove this just to keep it clean, unless you need to keep it for debugging

**Review by fameal — CHANGES_REQUESTED** [review]

File: scripts/findPath.js (hunk 3 above)
Comment (TBouder) [reply scripts/findPath.js:248]: This is needed to run get the path in the final file with this command : `node scripts/findPath.js > possiblesPaths.json`
The `>` will tell the system to write the 'logs' from the previous commands `node scripts/findPath.js` to the file `possiblesPaths.json`.

Comment (TBouder) [reply scripts/findPath.js:248]: But we could improve that, I will use another way

**Review by fameal — APPROVED** [review]
lgtm

**Pull request merged**